trick_source/trick_utils/comm/test/TCConnectTest.cpp

```diff
@@ -20,7 +20,7 @@ class TCConnectTest : public testing::Test {
          /* device */
          device = (TCDevice *) malloc(sizeof(TCDevice));
          memset( (void *)device,'\0',sizeof(TCDevice) );
-         device->hostname = "127.0.0.1";
+         device->hostname = (char*)"127.0.0.1";
```

Comment:
This is [not safe](https://arduino.stackexchange.com/a/13443). We should really make `hostname` a `const char*` in `TCDevice`. Barring that, we should use `strdup` here to make a truly mutable string.

Reply:
Hostname unfortunately can't be a const. I've tried it and const is discarded down the line when hostname is reassigned. I'll try using strdup and duplicate the literal into the char *.